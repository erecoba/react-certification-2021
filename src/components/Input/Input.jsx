import React from 'react';
import {
  Container,
  Inner,
  InputIcon,
  WrapperInput,
  TrayInput,
  InputStyle,
} from './Input.styled';

const Input = ({ icon, type = 'text', ...rest }) => {
  return (
    <Container>
      <Inner>
        {icon && <InputIcon>{icon}</InputIcon>}
        <WrapperInput>
          <TrayInput>
            <InputStyle {...rest} type={type} />
          </TrayInput>
        </WrapperInput>
      </Inner>
    </Container>
  );
};

export default Input;
