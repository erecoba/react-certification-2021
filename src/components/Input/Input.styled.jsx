import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  padding: 5px;
`;

const Inner = styled.div`
  display: flex;
  width: 100%;
`;

const InputIcon = styled.div`
  padding: 0 5px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const WrapperInput = styled.div`
  flex-grow: 1;
  margin-right: 1px;
`;

const TrayInput = styled.div`
  position: relative;
`;

const InputStyle = styled.input`
  padding: 5px 10px;
  margin: 2px;
  border: 2px solid transparent;
  background-color: transparent;
  height: 2rem;
  width: 100%;
  background-clip: padding-box;
  appearance: none;
  font-family: 'Roobert', sans-serif;
  color: ${({ theme }) => theme.label};

  border-radius: 8px;

  &:focus {
    outline: none;
    border-color: ${({ theme }) => theme.inputHoverBorder};
    background-color: ${({ theme }) => theme.inputHoverBg};
    color: ${({ theme }) => theme.colors.black};
  }
`;

export { Container, Inner, InputIcon, WrapperInput, TrayInput, InputStyle };
