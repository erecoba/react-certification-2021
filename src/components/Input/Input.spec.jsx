import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import Input from './Input';

describe('Avatar', () => {
  it('should render correctly inside button', () => {
    const theme = {
      colors: {
        white: '#fff',
      },
    };
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Input />
      </ThemeProvider>
    );

    expect(container.tagName).toMatchSnapshot('input');
  });
});
