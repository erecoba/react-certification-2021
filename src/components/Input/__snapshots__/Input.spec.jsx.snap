// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly inside button: input 1`] = `"DIV"`;
